### DISCUSSION TITLE
Change page build order

### DISCUSSION THREAD

**Discussion by fabio-nettis** [vercel/next.js#30094] [Help] [answered]
Silly question.

I have some code in my `index.tsx`'s getStaticProps that would need to run before any other pages are build. Is there any way to achieve that index.tsx is *ALWAYS* built first?

**Comment by gaston-flores** [comment, marked as answer]
I'm not sure there's a way to guarantee that. But why do you need a specific page's `getStaticProps` to run first? If you are generating some content for the other pages in `index.tsx`, maybe you could add an additional build step before running `next build`?

**Reply by fabio-nettis** [reply]
Yes have figured that out too by now. 🤣 Thanks anyway!
